### PR TITLE
feat: plugin settings pages & UI upgrades

### PR DESCRIPTION
**Pull request by nikosszzz**

### CODE REVIEW

File: src/api/PluginManager.ts [hunk 1]
```diff
@@ -43,6 +42,53 @@ export async function disablePlugin(plugin: string) {
...
+
+    try {
+        const confirm = await new Promise((resolve) => {
+            Dialog.show({
+                title: `Uninstall ${plugin}?`,
+                body: "Are you sure? This will stop and delete the plugin's files and cannot be undone.",
+                confirmText: "Uninstall",
+                cancelText: "Cancel",
+                confirmColor: "red",
+                isDismissable: false,
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:56]: why?

File: src/api/PluginManager.ts [hunk 2]
```diff
@@ -43,6 +42,53 @@ export async function disablePlugin(plugin: string) {
...
+                onCancel: () => resolve(false),
+            });
+        });
+
+        if (!confirm) return false;
+
+        logger.info(`Uninstalling plugin ${plugin}`);
+
+        const pluginInstance = plugins[plugin];
+        if (!pluginInstance.localPath) {
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:67]: just move this inside the try below you don't need to duplicate the dialog

File: src/api/PluginManager.ts [hunk 3]
```diff
@@ -74,6 +120,7 @@ export async function startCorePlugins() {
                 name,
                 description: "",
                 version: "1.0.0",
+                repo: "",
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:123]: link to rn repo instead

File: src/api/PluginManager.ts [hunk 4]
```diff
@@ -89,15 +136,15 @@ export async function startCorePlugins() {
 async function loadPlugin(pluginZip: string): Promise<Plugin | null> {
     if (plugins[pluginZip]) return plugins[pluginZip];
 
-    logger.info(`Loading plugin from ${pluginZip}.zip`);
+    logger.info(`Loading plugin from ${pluginZip}`);
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:139]: .zip is intentional

File: src/api/PluginManager.ts [hunk 5]
```diff
@@ -123,10 +170,12 @@ async function loadPlugin(pluginZip: string): Promise<Plugin | null> {
         // @ts-ignore
         loadedPlugin.pluginBuffer = pluginBuffer;
         loadedPlugin.enabled = false;
+        loadedPlugin.localPath = PLUGINS_DIRECTORY + pluginZip;
         plugins[manifest.name] = loadedPlugin;
 
         return loadedPlugin;
-    } catch (err) {
+    } catch (err: any) {
+        if (pluginName && plugins[pluginName]) plugins[pluginName].errors += err.stack;
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:178]:
```suggestion
        plugins[pluginName]?.errors += err?.stack ?? err;
```

File: src/api/PluginManager.ts [hunk 6]
```diff
@@ -148,7 +197,7 @@ async function startPlugin(plugin: string) {
         await loadedPlugin.start();
         loadedPlugin.enabled = true;
     } catch (err: any) {
-        loadedPlugin.errors = err.stack;
+        loadedPlugin.errors += err.stack;
```
Comment (rushiiMachine) [inline src/api/PluginManager.ts:200]:
```suggestion
        loadedPlugin.errors += err?.stack ?? err;
```

File: src/entities/Plugin.ts [hunk 7]
```diff
@@ -37,4 +39,10 @@ export class Plugin<SettingsSchema = any> {
         this.commands.unregisterAll();
         this.patcher.unpatchAll();
     }
+
+    /**
+     * This fetches the settings page that the plugin has defined.
+     * 
+     */
+    public getSettingsPage?(props: any[]): React.ReactElement;
```
Comment (rushiiMachine) [inline src/entities/Plugin.ts:47]: what is props in this instance

File: src/ui/PluginsPage.tsx [hunk 8]
```diff
@@ -65,31 +84,51 @@ const styles = Styles.createThemedStyleSheet({
...
+                                            if (!Users.getUser(a.id)) {
+                                                FetchUserActions.fetchProfile(a.id).then(() => {
+                                                    Profiles.showUserProfile({ userId: a.id });
+                                                });
+                                            } else {
+                                                Profiles.showUserProfile({ userId: a.id });
+                                            }
+                                        }}
+                                    >
+                                        {a.name}{i !== (plugin.authors as Author[]).length - 1 && <Text style={styles.text}>, </Text>}
```
Comment (rushiiMachine) [inline src/ui/PluginsPage.tsx:118]: just extract this into a variable it's unreadable

File: src/ui/ThemesPage.tsx [hunk 9]
```diff
@@ -111,40 +118,59 @@ function ThemeCard({ theme }: { theme: Theme; }) {
...
+                                            if (!Users.getUser(a.id)) {
+                                                FetchUserActions.fetchProfile(a.id).then(() => {
+                                                    Profiles.showUserProfile({ userId: a.id });
+                                                });
+                                            } else {
+                                                Profiles.showUserProfile({ userId: a.id });
+                                            }
+                                        }}
+                                    >
+                                        {a.name}{i !== (theme.authors as Author[]).length - 1 && <Text style={styles.text}>, </Text>}
```
Comment (rushiiMachine) [inline src/ui/ThemesPage.tsx:140]: same here

File: src/ui/patchSettings.tsx [hunk 10]
```diff
@@ -1,16 +1,16 @@
 import { sha } from "aliucord-version";
```
Comment (rushiiMachine) [inline src/ui/patchSettings.tsx:1]: can we move this file under somewhere else
like under src/patches 
\+ the themer/patcheTheme.tsx and then themerInit can go in root by itself

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

**Comment by amsyarasyiq** [comment]
major changes that have to be reviewed again:

- ui directory has been reorganized
	- `ui/components` and `ui/pages` has been created
	- some files have been moved into their respective folder
- stylesheets used across pages are now merged (export of `ui/components`)
- most of `PluginCard` and `ThemeCard` has been moved to `ui/components/Card`

**Review by rushiiMachine — APPROVED** [review]

**Pull request merged**